.github/workflows/todo-auto-issues.yml

```diff
@@ -0,0 +1,48 @@
+name: Check for TODOs
+
+on:
+  pull_request:
+    types:
+      - closed
+    branches:
+      - main
+
+jobs:
+  scan-for-todos:
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.merged == true
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        ref: ${{ github.event.pull_request.merge_commit_sha }}
+    
+    - name: Scan for TODOs
+      id: todos
+      run: |
+        git fetch origin main
+        todos=$(git diff ${{ github.event.pull_request.base.sha }}..${{ github.event.pull_request.merge_commit_sha }} --unified=0 | awk '/^\+.*TODO:/ { sub(/^.*TODO:\s*/, ""); sub(/^ */, ""); $0=toupper(substr($0,1,1)) substr($0,2); print $0 }')
```

Comment:
note: this line greps "TODO:"-patterned lines from a diff introduced by a merged pull request. It does that by only checking lines starting with `+` sign, since `git diff` marks added files with a plus in the output. Also, it removes all whitespaces after `:` and capitalizes a first non-whitespace character